src/reducers/project-state.js

```diff
@@ -160,20 +160,22 @@ const reducer = function (state, action) {
         }
         return state;
     case SET_PROJECT_ID:
-        // if we were already showing something, only fetch project if the
-        // project id has changed. This prevents re-fetching projects unnecessarily.
-        if (state.loadingState === LoadingState.SHOWING_WITH_ID) {
-            if (state.projectId !== action.id) {
+        // if we were already showing a project, and a different projectId is set, only fetch that project if
+        // projectId is a valid id, and projectId has changed. This prevents re-fetching projects unnecessarily.
+        if (action.id !== null && action.id !== '' && typeof action.id !== 'undefined') {
```

Comment:
The reducer is still doing some odd stuff here — it seems like this is a patch over a bug somewhere else. If we shouldn't be asking it to set the project id to something falsey, then why/where are we doing this?  I would want to know that I did that incorrectly when I did it, rather than being confused why nothing happened when I dispatched "set project id to null".

Reply:
Let's discuss this, there are a few ways to go.

Reply:
Made this change.